packages/telegram-cloud-storage-stamper/src/index.ts

```diff
@@ -0,0 +1,300 @@
+    } catch {
+      return false;
+    }
+  }
+
+  checkTelegramContext() {
+    if (window?.Telegram?.WebApp?.CloudStorage == null) {
+      throw new TelegramCloudStorageStamperError(
+        "Cannot use telegram stamper in non telegram mini-app environment, window.Telegram.WebApp.CloudStorage is not defined"
+      );
+    }
+  }
+
+  stringifyAPIKey(apiPublicKey: string, apiPrivateKey: string) {
+    return JSON.stringify({
+      apiPublicKey,
+      apiPrivateKey,
+    });
+  }
+
+  parseAPIKey(apiKey: string) {
+    try {
+      const parsedApiKey = JSON.parse(apiKey);
+
+      if (!this.isApiKey(parsedApiKey)) {
+        return {
+          apiPublicKey: "",
+          apiPrivateKey: "",
+        };
+      }
+
+      return {
+        apiPublicKey: parsedApiKey.apiPublicKey,
+        apiPrivateKey: parsedApiKey.apiPrivateKey,
+      };
+    } catch (err) {
+      throw new TelegramCloudStorageStamperError(
+        "Failed parsing API key from Telegram Cloud Storage"
+      );
+    }
+  }
+
+  isApiKey(apiKey: CloudStorageAPIKey) {
+    return (
+      typeof apiKey.apiPublicKey === "string" &&
+      typeof apiKey.apiPrivateKey === "string"
+    );
+  }
+
+  async getCloudStorageItem(key: string) {
+    return new Promise((resolve, reject) => {
+      window.Telegram.WebApp.CloudStorage.getItem(
+        key,
+        (err: any, value: string) => {
+          if (err != null || !value) {
+            reject(
+              new TelegramCloudStorageStamperError(
+                `Failed getting value: ${key} from Telegram Cloud Storage${
+                  err && `: ${err}`
+                }`
+              )
+            );
+          }
+
+          resolve(value);
+        }
+      );
+    });
+  }
+
+  async setCloudStorageItem(key: string, value: string) {
+    return new Promise((resolve, reject) => {
+      window.Telegram.WebApp.CloudStorage.setItem(
+        key,
+        value,
+        (err: any, stored: boolean) => {
+          if (err != null || !stored) {
+            reject(
+              new TelegramCloudStorageStamperError(
+                `Failed inserting value: ${value} into Telegram Cloud Storage at key: ${key}${
+                  err && `: ${err}`
+                }`
+              )
+            );
+          }
+
+          resolve(stored);
+        }
+      );
+    });
+  }
+
+  // clear key from telegram cloud storage
+  async clearCloudStorageItem(key: string) {
+    // check to see if we're in a telegram mini app context
+    this.checkTelegramContext();
+
+    await new Promise((resolve, reject) => {
+      window.Telegram.WebApp.CloudStorage.removeItem(
+        key,
+        (err: any, removed: boolean) => {
+          if (err || !removed) {
+            reject(
+              new TelegramCloudStorageStamperError(
+                `Failed removing key: ${key}${err && `: ${err}`}`
+              )
+            );
+          }
+
+          resolve(removed);
```

Comment:
you can simply `resolve()` without a boolean since you already check the boolean above. I think it's a cleaner interface?

Reply:
changed